### ISSUE TITLE
Remove / rework:UPnP use original port 

### ISSUE THREAD

**Issue by IX-285** [hrydgard/ppsspp#15630]
### What should happen

PPSSPP has option that opens both ports by port offset and original ones,however this feature not only doesn't work on neither device but is also misleading

There should be feature where ports would be translated so both PC and Phone devices work with PSP

*All options were tested on admin permissions between PSP > PC and Phone > PSP*

### Who would this benefit

Everyone with PSP hardware

### Games this would be useful in

Basically every game

### Checklist

- [X] Check the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already implemented.
- [X] [Search for other requests](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same feature.

**Comment by ANR2ME** [comment]
UPnP creates a port forwarding mapping between LAN_IP:LAN_PORT <--> PUBLIC_PORT (belonged to public IP).

"UPnP use Original Port (**Enabled = PSP compatibility**)" simply replace the PUBLIC_PORT with original port used by the game (ie. without Port Offset) so real PSP and non-PPSSPP emulators (those that use pro-online too) can communicate with PPSSPP over public IP.
_**It's a feature to be used when playing with real PSP with Android users that have port >= 1024 restriction**._
However, using original port have it's own drawbacks, due to many games are using ports < 100 which commonly used by internet services (ie. DBZ SB2 using port 23 commonly used for Telnet, port 25 commonly used for SMTP) thus, if someone sharing the same router with you have these kind of service running, you won't be able to use those ports.

When not using original port (which is the default), both LAN_PORT and PUBLIC_PORT where Port Offset being taken into account, which can avoid using commonly used/registered ports for internet services.

And, no PPSSPP won't open both ports (original and the one with port offset), only the PUBIC_PORT will be opened.
However, PPSSPP will keep track all the ports it have opened to avoid reopening them while PPSSPP is still running, those ports will be closed when PPSSPP being closed down.

The reason why PPSSPP need to minimize UPnP requests to the router is because on a slow router it could takes more than 500ms to open the port, and as i remembered there are games that will repeatedly create a socket -> bind -> listen (which triggered UPnP Add request) -> close the socket when not receiving anything within 500ms or smaller (which trigger UPnP Delete request), this kind of game will end up flooding your router with UPnP request(which is already slow), and when it failed the first time(ie. getting timed out while waiting for the port to be opened), the rest of the attempt will fails too.

By keeping tracks ports that have been opened by PPSSPP and not closing the port (yet) when the socket closed, if the 1st attempt were getting timed out, and the game try to open the port again on the next attempt it will most-likely a success and able to received incoming data, because the port has been opened before and didn't get closed when the game close the socket on the failed 1st attempt.

Anyway, it's just a simple port-forwarding feature, whether it worked or not depends on the game and whether there are running internet service on any device connected to the same router that might use the same port with the game.
And it should resulting the same to using Port Offset = 0, so if you can use Port Offset = 0 to play with real PSP over public IP, "Use Original Port" should also worked.

**Comment by unknownbrackets** [comment]
Given the above, I'm going to close.  If it's still an issue, leave a comment with appropriate information and we can reopen.

-[Unknown]

**Comment by MojoJojoDojo** [comment]
Yeah the option only matters for internet play mostly.
If you want to play over LAN you need to change the port offset to 0.
I made a change from it to 10000 on purpose because so many players on Android got issues with the default port 0.
And seems like having all games use 1 single port is still not coming anytime soon....

**Comment by ANR2ME** [comment]
Since some people like to enable all kind of settings without even checking what they're supposed to do first (just like how people like to enable all kind of graphic settings and end up getting glitches), i should probably show a popup or something asking the player to test Port Offset = 0 first when trying to enable "Use Original Port".
